Pcap++/header/PcapLiveDeviceList.h

```diff
@@ -21,10 +23,9 @@ namespace pcpp
 	 * devices are initialized on startup and wrap the network interfaces installed on the machine. This class enables access to them through
 	 * their IP addresses or get a vector of all of them so the user can search them in some other way
 	 */
-	class PcapLiveDeviceList
+	class PcapLiveDeviceList : public internal::DeviceListBase<PcapLiveDevice>
```

Comment:
If we make this class a container, maybe we also want to rename the other methods like:
- `getPcapLiveDeviceByIp` -> `getByIp`
- `getPcapLiveDeviceByName` -> `getByName`
- `getPcapLiveDeviceByIpOrName` -> `getByIpOrName`

We can deprecate the old names and later remove them.

Also, `getDnsServers()` doesn't seems to really belong here, but I don't know what to do with it...

Reply:
Sure. Tbh, it might even be possible to get some those into the base class with some template magic fuckery, as some are shared (LiveDevice and RemoteDevice basically share the same getter methods).

Reply:
Changed the functions to syntax `getDeviceBy*`. It is shorter and `Dpdk` and `Kni` lists already used that syntax.

Related commits: a73d4b9, ffac432, 151f09f.